### PR TITLE
verify the common book genre that’s being borrowed- ZU

### PR DESCRIPTION
**Pull request by untuzina**
Please review! Thank you!

### CODE REVIEW

**Review by Mariagirbu — APPROVED** [review]

File: src/test/java/com/library/runner/CukesRunner.java [hunk 1]
```diff
@@ -15,7 +15,7 @@
         features = "src/test/resources/features" ,
         glue = "com/library/steps",
         dryRun = false,
-        tags = ""
+        tags = "@db"
```
Comment (farukgoc) [inline src/test/java/com/library/runner/CukesRunner.java]: Please remove @db tag from here

File: src/test/java/com/library/steps/B28G31_148_StepDefs_BorrowedTheMost_ZU.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.junit.Assert;
+
+
+
+public class B28G31_148_StepDefs_BorrowedTheMost_ZU {
+
+    LoginPage loginPage = new LoginPage();
```
Comment (farukgoc) [inline src/test/java/com/library/steps/B28G31_148_StepDefs_BorrowedTheMost_ZU.java]: This is redundant. Your user story does not require login page object

File: src/test/java/com/library/steps/B28G31_148_StepDefs_BorrowedTheMost_ZU.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+
+    @Given("the {string} on the home page ZU")
+    public void the_on_the_home_page_ag(String userType) {
+
+        loginPage.login(userType);
+
+
+    }
```
Comment (farukgoc) [inline src/test/java/com/library/steps/B28G31_148_StepDefs_BorrowedTheMost_ZU.java]: You dont have this step in your feature file. Why did you add it here. The user does not need to login to the application in your user story

File: src/test/java/com/library/steps/B28G31_148_StepDefs_BorrowedTheMost_ZU.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+
+    @When("I execute query to find most popular book genre - ZU")
+    public void i_execute_query_to_find_most_popular_book_genre_zu() {
+    DB_Util.runQuery("SELECT bc.name, COUNT(*)\n" +
+            "FROM book_borrow bb\n" +
+            "         INNER JOIN books b ON bb.book_id = b.id\n" +
+            "         INNER JOIN book_categories bc ON b.book_category_id = bc.id\n" +
+            "GROUP BY bc.name\n" +
+            "ORDER BY count(*) DESC;");
```
Comment (farukgoc) [inline src/test/java/com/library/steps/B28G31_148_StepDefs_BorrowedTheMost_ZU.java:29]: You need to assign the result from this query to actualGenre

File: src/test/java/com/library/steps/B28G31_148_StepDefs_BorrowedTheMost_ZU.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    }
+
+
+    @Then("verify {string} is the most popular book genre. - ZU")
+    public void verify_is_the_most_popular_book_genre_zu(String expectedGenre) {
+        String actualGenre = "Fantasy";
```
Comment (farukgoc) [inline src/test/java/com/library/steps/B28G31_148_StepDefs_BorrowedTheMost_ZU.java]: This is harcoding. It is not acceptable

File: src/test/java/com/library/steps/Hooks.java [hunk 6]
```diff
@@ -13,7 +13,7 @@
 
 public class Hooks {
 
-    @Before ("@ui")
```
Comment (farukgoc) [inline src/test/java/com/library/steps/Hooks.java]: You should not remove this tag. This will fail the others step definitions

File: src/test/java/com/library/steps/Hooks.java [hunk 7]
```diff
@@ -23,7 +23,7 @@ public void setUp(){
 
     }
 
-    @After ("@ui")
```
Comment (farukgoc) [inline src/test/java/com/library/steps/Hooks.java]: You should not remove this tag

File: src/test/resources/features/B28G31-147-OFG.feature [hunk 8]
```diff
@@ -1,4 +1,4 @@
-@B28G31-153 @ui
```
Comment (farukgoc) [inline src/test/resources/features/B28G31-147-OFG.feature]: Why did you remove this. You should not touch someone else's feature file

File: src/test/resources/features/B28G31-147-OFG.feature [hunk 9]
```diff
@@ -8,7 +8,7 @@ Feature: Default
 		
 
 	#As a data consumer, I want UI and DB book information are match.
-	@B28G31-152 @db
```
Comment (farukgoc) [inline src/test/resources/features/B28G31-147-OFG.feature]: You should not remove this either

File: src/test/resources/features/B28G31-150_BG.feature [hunk 10]
```diff
@@ -8,4 +8,4 @@ Feature: Books module
     And the user searches for "Bobby Book" book - BG
     When the user clicks Borrow Book - BG
     Then verify that book is shown in "Borrowing Books" page - BG
-    And  verify logged student has same book in database - BG
```
Comment (farukgoc) [inline src/test/resources/features/B28G31-150_BG.feature]: Why did you change here. It will affect Bobby's step definitions

File: src/test/resources/features/B28G31_148_ZU.feature [hunk 11]
```diff
@@ -0,0 +1,11 @@
+@B28G31-164  @db
+Feature: verify the common book genre that’s being borrowed- ZU
+
+	@B28G31-163  @db
+	Scenario: verify the common book genre that’s being borrowed- ZU
+		Feature: As a data consumer, I want to know genre of books are being borrowed the most - ZU
```
Comment (farukgoc) [inline src/test/resources/features/B28G31_148_ZU.feature]: Why do you have 2 Feature keywords

**Review by farukgoc — COMMENTED** [review]
We can talk about this comments tomorrow at the meeting

**Review by farukgoc — APPROVED** [review]
Good job Zina

**Review by Alexa3462 — APPROVED** [review]
Looks good!
